src/compiler/utilities.ts

```diff
@@ -3504,6 +3504,43 @@ namespace ts {
     export function getCombinedLocalAndExportSymbolFlags(symbol: Symbol): SymbolFlags {
         return symbol.exportSymbol ? symbol.exportSymbol.flags | symbol.flags : symbol.flags;
     }
+
+    export function isWriteOnlyAccess(node: Node) {
+        return writeKind(node) === WriteKind.Write;
+    }
+
+    export function isWriteAccess(node: Node) {
+        return writeKind(node) !== WriteKind.Read;
+    }
+
+    /**
+     * @param writeOnly If set, return true for `x++;` but not for `f(x++);`, which uses `x` in addition to writing to it.

```

Comment:
I don't think this comment is supposed to be here anymore; but, this is much nicer now. Would `AccessKind` be better than `WriteKind`, considering it can indicate a read?